src/runtime/dict.cpp

```diff
@@ -31,25 +31,57 @@ using namespace pyston::ExceptionStyle;
 using pyston::ExceptionStyle::ExceptionStyle;
 
 Box* dictRepr(BoxedDict* self) {
-    std::vector<char> chars;
-    chars.push_back('{');
-    bool first = true;
-    for (const auto& p : self->d) {
-        if (!first) {
-            chars.push_back(',');
+
+    try {
```

Comment:
@kmod i think that i have some misunderstanding about exception mechanism... in conclusion you mean try catch block only need when Py_ReprEnter failed or call repr()??